docs/source/integrations/plugins-event-reference.mdx

```diff
@@ -98,69 +109,83 @@ const server = new ApolloServer({
 })
 ```
 
+</MultiCodeBlock>
+
 ### `renderLandingPage`
 
-This event enables you to serve a [custom landing page](../testing/build-run-queries/#custom-landing-page) from Apollo Server's base URL. The event is fired once by Apollo Server after all `serverWillStart` events run. **At most one installed plugin can define a `renderLandingPage` handler.** Otherwise, Apollo Server throws an error on startup.
+This event enables you to serve a [custom landing page](../workflow/build-run-queries/#custom-landing-page) from Apollo Server's base URL. The event is fired once by Apollo Server after all `serverWillStart` events run. **At most one installed plugin can define a `renderLandingPage` handler.** Otherwise, Apollo Server throws an error on startup.
 
 You define your plugin's `renderLandingPage` handler in the object returned by your [`serverWillStart`](#serverwillstart) handler, which enables it to read values passed to `serverWillStart`:
 
-```js {5-22} title="index.js"
-const server = new ApolloServer({
-  // ... other configuration ...
+<MultiCodeBlock>
 
+```ts title="index.ts"
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
   plugins: [
     {
       async serverWillStart() {
         return {
+          //highlight-start
           async renderLandingPage() {
             const html = `
-<!DOCTYPE html>
-<html>
-  <head>
-  </head>
-  <body>
-    <h1>Hello world!</h1>
-  </body>
-</html>`;
+              <!DOCTYPE html>
+              <html>
+                  <head>
+                  </head>
+                  <body>
+                      <h1>Hello world!</h1>
+                  </body>
+              </html>`;
             return { html };
-          }
-        }
-      }
-    }
-  ]
+          },
+          //highlight-end
+        };
+      },
+    },
+  ],
 });
 ```
 
+</MultiCodeBlock>
+
 The handler should return an object with a string `html` field. The value of that field is served as HTML for any requests with `accept: text/html` headers.
 
-For more landing page options, see [Changing the landing page](../testing/build-run-queries/#changing-the-landing-page).
+For more landing page options, see [Changing the landing page](../workflow/build-run-queries/#changing-the-landing-page).
 
 #### Example
 
-```js
+<MultiCodeBlock>
+
+```ts
 const server = new ApolloServer({
   /* ... other necessary configuration ... */
 
-  plugins: [
+   plugins: [
     {
       async serverWillStart() {
         return {
           async renderLandingPage() {
             return { html: `<html><body>Welcome to your server!</body></html>` };
-          }
-        }
-      }
-    }
-  ]
+          },
+        };
+      },
+    },
+  ],
 })
 ```
 
+</MultiCodeBlock>
+
 
 ### `requestDidStart`
 
+> **New in Apollo Server 4**: In Apollo Server 4, `requestDidStart` hooks are called in parallel rather than in series.
+
 The `requestDidStart` event fires whenever Apollo Server begins fulfilling a GraphQL request.
 
-```typescript
+```ts
```

Comment:
@glasser and @trevor-scheer, in the interest of time, I tried updating each of these function signatures instead of making a table for `GraphQLRequestContext`. I'll add that to my list of todos after finishing up these AS3 -> AS4 docs PRs. 

I tried my best to read the types here, but could one of you carefully double-check these signatures for me just to make sure I got all the fields right? Thank you in advance! 🙇

Reply:
I finished my first pass but didn't look at these yet, just FYI

Reply:
@rkoron007 thanks for flagging this. I pushed up [0ee1750](https://github.com/apollographql/apollo-server/pull/6873/commits/0ee175076bb528c927428e8f91f22c55ac131591) with some changes for these.

Notably, `metrics` is no longer optional so we don't need to add it to all the `WithRequired` lists. I also reordered them to be consistent amongst themselves and ordered logically (by when things are expected to be on the request context, i.e. `source` and `queryHash` are resolved first, followed by `document`, then `operationName`, etc.).

Reply:
@glasser I updated the `externalTypes` with this commit, if you could give this a sanity check when you're back that'd be great.